apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/schemas/Sgv2IndexesResourceImpl.java

```diff
@@ -80,23 +80,15 @@ public Uni<RestResponse<Map<String, Object>>> addIndex(
   @Override
   public Uni<RestResponse<Void>> deleteIndex(
       String keyspaceName, String tableName, String indexName, boolean ifExists) {
+
+    // Use 'queryWithTableAsync' to check existence of Keyspace and Table as well as access;
+    // existence and access to Index itself verified by persistence backend
     return queryWithTableAsync(
             keyspaceName,
             tableName,
             true,
-            table -> {
-              if (!ifExists
-                  && table.getIndexesList().stream()
-                      .noneMatch(i -> indexName.equals(i.getName()))) {
-                throw new WebApplicationException(
-                    String.format("Index '%s' not found.", indexName), Status.NOT_FOUND);
-              }
-              return new QueryBuilder()
-                  .drop()
-                  .index(keyspaceName, indexName)
-                  .ifExists(ifExists)
-                  .build();
-            })
+            table ->
```

Comment:
but this should still ensure table does exist before firing that query?

Reply:
But table existence is not verified, see that null check for `table` that was on line `91`? Fix this and lets merge this?

Reply:
> But table existence is not verified, see that null check for `table` that was on line `91`? Fix this and lets merge this?

Ok thanks I'll have a look there. I forgot that checking for `Table` not being null is left to handlers, whereas keyspace non-existence is

Reply:
Adding a test to verify, but Table existence is actually checked in helper method in `RestResourceBase`, so that is covered. But adding a test for this makes sense, good suggestion.